File: examples/capture.py
import json
import os
import sys
import urllib.error
import urllib.request


if len(sys.argv) not in (2, 3) or not os.environ.get("LATCHSHOT_API_KEY"):
    raise SystemExit(
        "Usage: LATCHSHOT_API_KEY=... python3 examples/capture.py URL [OUTPUT.png]"
    )

target = sys.argv[1]
output = sys.argv[2] if len(sys.argv) == 3 else "capture.png"
request = urllib.request.Request(
    "https://latchshot.fly.dev/v1/render",
    data=json.dumps(
        {"url": target, "format": "png", "width": 1440, "height": 900}
    ).encode(),
    headers={
        "Authorization": f"Bearer {os.environ['LATCHSHOT_API_KEY']}",
        "Content-Type": "application/json",
    },
    method="POST",
)

try:
    with urllib.request.urlopen(request, timeout=45) as response:
        body = response.read()
        with open(output, "xb") as artifact:
            artifact.write(body)
        print(
            json.dumps(
                {
                    "output": output,
                    "bytes": len(body),
                    "renderMs": response.headers.get("X-Latchshot-Render-Ms"),
                    "remaining": response.headers.get("X-Quota-Remaining"),
                }
            )
        )
except urllib.error.HTTPError as error:
    raise SystemExit(
        f"Latchshot returned HTTP {error.code}: {error.read().decode(errors='replace')}"
    ) from error

File: examples/capture.mjs
import { writeFile } from 'node:fs/promises';

const [url, output = 'capture.png'] = process.argv.slice(2);
const apiKey = process.env.LATCHSHOT_API_KEY;

if (!url || !apiKey) {
  console.error('Usage: LATCHSHOT_API_KEY=... node examples/capture.mjs URL [OUTPUT.png]');
  process.exit(2);
}

const response = await fetch('https://latchshot.fly.dev/v1/render', {
  method: 'POST',
  headers: {
    authorization: `Bearer ${apiKey}`,
    'content-type': 'application/json',
  },
  body: JSON.stringify({ url, format: 'png', width: 1440, height: 900 }),
  signal: AbortSignal.timeout(45_000),
});

if (!response.ok) {
  throw new Error(`Latchshot returned HTTP ${response.status}: ${await response.text()}`);
}

const bytes = new Uint8Array(await response.arrayBuffer());
await writeFile(output, bytes, { flag: 'wx' });
console.log(JSON.stringify({
  output,
  bytes: bytes.byteLength,
  renderMs: response.headers.get('x-latchshot-render-ms'),
  remaining: response.headers.get('x-quota-remaining'),
}));
